subprojects/model-core/src/test/groovy/org/gradle/internal/reflect/validation/ValidationMessageCheckerTest.groovy

```diff
@@ -0,0 +1,804 @@
+
+Possible solution: Configure 'output' to point to a file, not a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+
+        when:
+        render cannotWriteToFile {
+            type('Writer').property('output')
+            file(location)
+            ancestorIsNotDirectory(ancestor)
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'Writer' property 'output' is not writable because '${location}' ancestor '${ancestor}' is not a directory.
+
+Reason: Cannot write a file to a location pointing at a directory.
+
+Possible solution: Configure 'output' to point to a file, not a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.CANNOT_WRITE_TO_RESERVED_LOCATION
+    )
+    def "tests output of cannotWriteToReservedLocation"() {
+        def reserved = dummyLocation()
+
+        when:
+        render cannotWriteToReservedLocation {
+            type('SomeTransform')
+            forbiddenAt(reserved)
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'SomeTransform' points to '${reserved}' which is managed by Gradle.
+
+Reason: Trying to write an output to a read-only location which is for Gradle internal use only.
+
+Possible solution: Select a different output location.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_to_reserved_location for more details about this problem.
+"""
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.UNSUPPORTED_NOTATION
+    )
+    def "tests output of unsupportedNotation"() {
+        when:
+        render unsupportedNotation {
+            type('Human').property('face')
+            value('visible', 'person')
+            candidates("a mask", "a vaccine")
+            cannotBeConvertedTo('safe person')
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'Human' property 'face' has unsupported value 'visible'.
+
+Reason: Type 'person' cannot be converted to a safe person.
+
+Possible solutions:
+  1. Use a mask.
+  2. Use a vaccine.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#unsupported_notation for more details about this problem.
+"""
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.INVALID_USE_OF_CACHEABLE_ANNOTATION
+    )
+    def "tests output of invalidUseOfCacheableAnnotation"() {
+        when:
+        render invalidUseOfCacheableAnnotation {
+            type('SomeTransform')
+            invalidAnnotation('CacheableTask')
+            onlyMakesSenseOn('Task')
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'SomeTransform' Using @CacheableTask here is incorrect.
```

Comment:
Something is wrong here. Either `Using` should be lower case or there is a `:` missing after `SomeTransform`.

Reply:
Yes I'll update the message to make it nicer.